build.sbt

```diff
@@ -151,19 +154,40 @@ lazy val iam = project
                                 scalaTest   % Test)
   )
 
-lazy val shaclValidator = project
-  .in(file("modules/ld/shacl-validator"))
-  .dependsOn(types)
+lazy val indexTypes = project
+  .in(file("modules/index/types"))
   .settings(
-    name                := "shacl-validator",
-    moduleName          := "shacl-validator",
-    resolvers           += Resolver.bintrayRepo("bogdanromanx", "maven"),
-    libraryDependencies ++= Seq(journal, wesoSchema, catsCore, circeCore, circeParser % Test, scalaTest % Test)
+    name                := "commons-index-types",
+    moduleName          := "commons-index-types",
+    libraryDependencies ++= Seq(akkaHttp, catsCore, circeCore, scalaTest % Test, circeGenericExtras % Test)
+  )
+
+lazy val elasticClient = project
+  .in(file("modules/index/elastic-client"))
+  .dependsOn(http, indexTypes, test)
+  .settings(
+    name       := "elastic-client",
+    moduleName := "elastic-client",
+    libraryDependencies ++= Seq(
+      akkaStream,
+      circeCore,
+      circeOptics,
```

Comment:
to be honest pulling in transiently monocle and scalaz just for the decoders seems a little heavy...

Reply:
I'll try to decode it without circeOptics even though is quite useful